models.go

```diff
@@ -0,0 +1,84 @@
+package flagsmithapi
+
+import (
+	"encoding/json"
+	"log"
+)
+
+type FeatureStateValue struct {
+	Type         string  `json:"type"`
+	StringValue  *string `json:"string_value"`
+	IntegerValue *int64  `json:"integer_value"`
+	BooleanValue *bool   `json:"boolean_value"`
+}
+
+type FeatureState struct {
+	ID                int64              `json:"id"`
+	FeatureStateValue *FeatureStateValue `json:"feature_state_value"`
+	Enabled           bool               `json:"enabled"`
+	Feature           int64              `json:"feature"`
+	Environment       int64              `json:"environment"`
+}
+
+func (fs *FeatureState) UnmarshalJSON(data []byte) error {
+	var obj struct {
+		ID                int64           `json:"id"`
+		FeatureStateValue json.RawMessage `json:"feature_state_value"`
+		Enabled           bool            `json:"enabled"`
+		Feature           int64           `json:"feature"`
+		Environment       int64           `json:"environment"`
+	}
+
+	err := json.Unmarshal(data, &obj)
+
+	if err != nil {
+		return err
+	}
+	fs.ID = obj.ID
+	fs.Enabled = obj.Enabled
+	fs.Feature = obj.Feature
+	fs.Environment = obj.Environment
+
+	// If the feature state value is a struct then unmarshal, set
```

Comment:
Can we explain the scenario here where FSV is / isn't a struct? I'm a bit lost on this.

Reply:
Yeah, I added the comment